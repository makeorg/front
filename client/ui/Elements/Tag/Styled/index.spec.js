import React from 'react';
import renderer from 'react-test-renderer';
import { TagStyle } from './index';

describe('ComponentStyle', () => {
  it('snapshot by default', () => {
    const component = renderer.create(<TagStyle>Hello</TagStyle>).toJSON();
    expect(component).toMatchSnapshot();
  });

  it('must set color from props', () => {
    const component = renderer
      .create(
        <TagStyle color="black" backgroundColor="blue">
          Hello
        </TagStyle>
      )
      .toJSON();
    expect(component).toHaveStyleRule('color', 'black');
    expect(component).toHaveStyleRule('background-color', 'blue');
  });
});
