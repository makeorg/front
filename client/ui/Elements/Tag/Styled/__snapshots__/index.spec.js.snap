// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ComponentStyle snapshot by default 1`] = `
.c0 {
  position: relative;
  cursor: pointer;
  padding: 0 10px 0 12px;
  -webkit-text-decoration: none;
  text-decoration: none;
  height: 26px;
  line-height: 24px;
  font-size: 14px;
  border: none;
  margin-left: 12px;
  margin-right: 12px;
  margin-bottom: 12px;
}

.c0:hover:before {
  border-color: transparent transparent transparent;
}

.c0:before {
  content: '';
  position: absolute;
  top: 0;
  left: -12px;
  width: 0;
  height: 0;
  border-color: transparent transparent transparent;
  border-style: solid;
  border-width: 12px 12px 14px 0;
}

.c0:after {
  content: '';
  position: absolute;
  top: 10px;
  left: 0;
  float: left;
  width: 6px;
  height: 6px;
  border-radius: 3px;
  background: #fff;
}

<button
  className="c0"
>
  Hello
</button>
`;
